src/core/data/particles/particle_array.hpp

```diff
@@ -233,6 +233,17 @@ class ParticleArray
 
     auto& box() const { return box_; }
 
+
+    auto& update_from(This const& that)
```

Comment:
I know it's not from this PR, but I'm curious regarding the need to define `This` actually, we could just use `ParticleArray` (without the template specialization) that would be clearer.

```cpp

template<typename T>
struct A
{
  auto& update_from(A const& from)
  {
      std::cout << __LINE__ << "\n";
      return *this;
  }
};
```